backend/src/main/java/capstone/backend/services/OrderToSupplierService.java

```diff
@@ -45,43 +50,63 @@ public OrderToSupplierDTO createOrder(OrderToSupplierDTO order) throws EntityNot
         return mapOrder(repo.save(mapOrder(order)));
     }
 
-    private boolean checkIfProductsExistent(OrderToSupplierDTO order) {
+    public OrderToSupplierDTO receiveOrder(OrderToSupplierDTO order, OrderStatus status) throws EntityNotFoundException, IllegalArgumentException {
+        if (status != RECEIVED) {
+            throw new IllegalArgumentException("We couldn't process your request!");
+        }
+        if (!orderExists(order)) {
+            throw new EntityNotFoundException("The order you're trying to receive doesn't exist");
+        }
+        if(order.getStatus() == RECEIVED){
+            throw new IllegalArgumentException("The order you're trying to receive has already been received");
+        }
+        productService.adjustAmountInStock(order.getOrderItems());
+        repo.findById(order.getId()).ifPresent(receivedOrder -> receivedOrder.setStatus(RECEIVED));
+        return mapOrder(repo.findById(order.getId()).orElseThrow());
+    }
+
+    private boolean orderExists(OrderToSupplierDTO order) {
+        return (order.getId() != null && repo.existsById(order.getId()));
+    }
+
+    private void productsExist(OrderToSupplierDTO order, String message) {
+        if (!productsExist(order)) {
+            throw new IllegalArgumentException(message);
+        }
+    }
+
+    private void validateNewOrder(OrderToSupplierDTO order) throws IllegalArgumentException, EntityWithThisIdAlreadyExistException {
+        productsExist(order, "You tried to order a product that doesn't exist!");
+        if (!supplierService.supplierExists(order.getSupplier().getId())) {
+            throw new IllegalArgumentException("You tried to order from a supplier that doesn't exist!");
+        }
+        if (supplierDoesNotCarryProduct(order)) {
+            throw new IllegalArgumentException("The supplier doesn't carry one or several of the items you tried to order!");
```

Comment:
"The supplier doesn't carry at least one of the items you tried to order!"
"At least one of the items you tried to order is out of stock"

Reply:
no, it's not about an item being out of stock. it's a failsafe in case somehow the data got manipulated on the way from frontend to backend and you have an item on your order that is not actually carried by this supplier